shm/island.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <string>
+
+class Coordinates {
+public:
+    Coordinates(size_t, size_t);
+    bool operator==(const Coordinates& pos) const {
+        return positionX == pos.positionX && positionY == pos.positionY;
+    }
+private:
+    size_t positionX;
+    size_t positionY;
```

Comment:
I know that in issue description was "positionX" and "positionY", but what do you think about changing names to "positionX_" and "positionY_"? But it's up to you.